cmd/aperture-agent/agent/otel-config.go

```diff
@@ -89,7 +86,7 @@ func provideAgent(
 				ims[fmt.Sprintf("%s/%s", prefix, k)] = v
 			}
 		}
-		if err := inframeter.AddInfraMeters(otelCfg, ims); err != nil {
+		if err := inframeter.AddInfraMeters(otelCfg, tc.GetPolicyName(), tc.GetTelemetryCollectorId(), ims); err != nil {
```

Comment:
This is now incorrect:

```
		otelCfg := baseOtelCfg.MustCopy()
		ims := map[string]*policylangv1.InfraMeter{}
		for prefix, v := range allInfraMeters {
			for k, v := range v {
				ims[fmt.Sprintf("%s/%s", prefix, k)] = v
			}
		}
		if err := inframeter.AddInfraMeters(otelCfg, tc.GetPolicyName(), tc.GetTelemetryCollectorId(), ims); err != nil {
			log.Error().Err(err).Msg("unable to add custom metrics pipelines")
			utils.Shutdown(shutdowner)
			return
		}
```

`ims` contain all inframeters – not only from current policy.

Reply:
Added a fix for this